subprojects/core-api/src/main/java/org/gradle/caching/MapBasedBuildCacheService.java

```diff
@@ -1,66 +0,0 @@
-/*
- * Copyright 2016 the original author or authors.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.gradle.caching;
-
-import org.gradle.internal.io.StreamByteBuffer;
-
-import java.io.ByteArrayInputStream;
-import java.io.IOException;
-import java.util.concurrent.ConcurrentMap;
-
-/**
- * Simple build cache implementation that delegates to a {@link ConcurrentMap}.
- *
- * @since 3.5
- */
-public class MapBasedBuildCacheService implements BuildCacheService {
```

Comment:
Why remove this in this PR? `MapBasedBuildCacheService` is part of the public API and is used by the HazelCast plugin IIRC.

I'm not saying we should keep it, but if we are to remove it, then we should remove it via deprecation.

Reply:
I can remove this change from the PR.

I thought this became dead code when we no longer allowed other implementations of the local build cache?

How do you load one of these?

Reply:
We never stopped allowing other implementations. You can register them via `BuildCacheConfiguration.registerBuildCacheService()`.